### ISSUE TITLE
Terraform requires arguments that are optional when creating aws_iot_topic_rule

### ISSUE THREAD

**Issue by jeandek** [terraform-providers/terraform-provider-aws#6195]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.8

### Affected Resource(s)

* aws_iot_topic_rule

### Terraform Configuration Files

```hcl
resource "aws_iot_topic_rule" "client_id_mapping" {
  name        = "${var.env_name}_client_id_mapping"
  enabled     = "true"

  sql = <<EOF
SELECT clientId, {"key": "value"} as mapping
FROM '$$aws/events/subscriptions/subscribed/#' 
EOF

  sql_version = "2016-03-23"

  dynamodb {
    hash_key_field  = "client_id"
    hash_key_value  = "$${clientId}"
    hash_key_type   = "STRING"
    range_key_field = ""
    range_key_value = ""
    payload_field   = "$${mapping}"
    table_name      = "${var.env_name}_client_id_mapping"
    role_arn        = "${aws_iam_role.client_id_mapping_role.arn}"
  }
}
```

### Debug Output

```bash
>>> terraform apply --var-file=../../../variables/dev.tfvar
Error: aws_iot_topic_rule.client_id_mapping: "dynamodb.0.range_key_field": required field is not set
Error: aws_iot_topic_rule.client_id_mapping: "dynamodb.0.range_key_value": required field is not set
```

### Expected Behavior

These range_key arguments of the DynamoDB action are optional according to the AWS API reference and the resource's TF documentation.

### Actual Behavior

Both arguments are marked as required in the [source code](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_iot_topic_rule.go).

### References

* [iot_topic_rule TF docs](https://www.terraform.io/docs/providers/aws/r/iot_topic_rule.html)
* [CreateTopicRule API Reference](https://docs.aws.amazon.com/iot/latest/apireference/API_CreateTopicRule.html)
* [DynamoDBAction API Reference]( https://docs.aws.amazon.com/iot/latest/apireference/API_DynamoDBAction.html)

**Comment by jeandek** [comment]
The fix is probably to set "Optional: true" in the source code for both arguments, but I'm not comfortable enough with Go and Terraform to write tests for it...

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.59.0 of the Terraform AWS Provider, likely in the next day or two. 👍
